src/doprnt.c

```diff
@@ -503,7 +503,7 @@ esprintf (char *buf, char const *format, ...)
   return nbytes;
 }
 
-#if HAVE_MODULES || (defined HAVE_X_WINDOWS && defined USE_X_TOOLKIT)
+#if HAVE_MODULES
```

Comment:
This looks wrong to me. Shouldn't we still consider `defined HAVE_X_WINDOWS`?

Reply:
I understand your confusions. The problem is that `GTK` is not an `X_TOOLKIT` in Emacs' logic.

* If we build with `gtk3`
  * `HAVE_WINDOW_SYSTEM` is true
  * `HAVE_X_WINDOWS` is **true**
  * `HAVE_X_11` is true
  * `USE_X_TOOLKIT` is **false**
  * `USE_GTK` is true
  * `HAVE_GTK3` is true

* If we build with `lucid`
  * `HAVE_WINDOW_SYSTEM` is true
  * `HAVE_X_WINDOWS` is **true**
  * `HAVE_X_11` is true
  * `USE_X_TOOLKIT` is **true**
  * `USE_LUCID` is true

In the vanilla Emacs, using `gtk3` toolkit doesn't trigger the compilation of those codes.
If we change the line to `#if HAVE_MODULES || defined HAVE_X_WINDOWS`, those codes will be compiled when using `gtk3` toolkit (which isn't expected).

Reply:
Are you sure about that:

```
#### Report on what we decided to do.
#### Report GTK as a toolkit, even if it doesn't use Xt.
#### It makes printing result more understandable as using GTK sets
#### toolkit_scroll_bars to yes by default.
if test "${HAVE_GTK}" = "yes"; then
  USE_X_TOOLKIT="$USE_GTK_TOOLKIT"
fi
```

Reply:
Yes, I am sure.

I checked the config.h after building with the multiple toolkits.  You can check it too.

The lines you show here is just correct the `printing result` of configure.ac. It doesn't set the `USE_X_TOOLKIT` macro to `true` for c codes.

Reply:
This is a great example of why I am happy to see us cut off the old code.

Reply:
Then the Xaw3d should go too as it depends on X_Toolkit

Reply:
Deleted Xaw3d.